components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanService.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash.service
+
+import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.GleanMetrics.CrashMetrics
+
+/**
+ * A [CrashReporterService] implementation for recording metrics with Glean
+ */
+class GleanService : CrashReporterService {
+    override fun report(crash: Crash.UncaughtExceptionCrash) {
+        CrashMetrics.uncaughtException.add()
+    }
+
+    override fun report(crash: Crash.NativeCodeCrash) {
+        CrashMetrics.nativeCodeCrash.add()
+    }
```

Comment:
I think there's a potential problem here: The crash reporter is running in a separate process (since the app process may have died). Glean is not initialized in this other process and probably shouldn't be in other's than the main app process.

Reply:
Will test to see how this works out..  thanks for the heads up.

Reply:
Finally found a solution that didn't break Glean or lib-crash 😄